tests/Kernel/TimelineTest.php

```diff
@@ -0,0 +1,199 @@
+    'oe_content',
+    'oe_content_timeline_field',
+    'rdf_entity',
+    'rdf_skos',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->setUpSparql();
+    $this->installEntitySchema('node');
+    $this->installConfig([
+      'field',
+      'node',
+      'filter',
+      'rdf_entity',
+    ]);
+    $this->installSchema('user', 'users_data');
+
+    $this->installEntitySchema('rdf_entity');
+    $this->installEntitySchema('skos_concept');
+    $this->installEntitySchema('skos_concept_scheme');
+
+    $this->container->get('theme_installer')->install(['oe_theme']);
+    $this->container->get('theme_handler')->setDefault('oe_theme');
+    $this->container->set('theme.registry', NULL);
+
+    // Create content type.
+    $type = NodeType::create(['name' => 'Test content type', 'type' => 'test_ct']);
+    $type->save();
+
+    FilterFormat::create([
+      'format' => 'my_text_format',
+      'name' => 'My text format',
+      'filters' => [
+        'filter_autop' => [
+          'module' => 'filter',
+          'status' => TRUE,
+        ],
+      ],
+    ])->save();
+
+    // Add text field to entity, to sort by.
+    $fieldStorage = FieldStorageConfig::create([
+      'field_name' => 'field_timeline',
+      'entity_type' => 'node',
+      'type' => 'timeline_field',
+      'cardinality' => -1,
+      'entity_types' => ['node'],
+    ]);
+    $fieldStorage->save();
+
+    $field = FieldConfig::create([
+      'label' => 'Timeline field',
+      'field_name' => 'field_timeline',
+      'entity_type' => 'node',
+      'bundle' => 'test_ct',
+      'settings' => [],
+      'required' => FALSE,
+    ]);
+    $field->save();
+
+    EntityViewDisplay::create([
+      'targetEntityType' => $field->getTargetEntityTypeId(),
+      'bundle' => $field->getTargetBundle(),
+      'mode' => 'default',
+      'status' => TRUE,
+    ])->setComponent($fieldStorage->getName(), $this->displayOptions)
+      ->save();
+  }
+
+  /**
+   * Test the timeline field rendering with formatter and ecl template.
+   */
+  public function testTimelineRender(): void {
+    $values = [
+      'type' => 'test_ct',
+      'title' => 'My node title',
+      'field_timeline' => [
+        [
+          'label' => '16/07/2019',
+          'title' => 'Item 1',
+          'body' => 'Item 1 body',
+          'format' => 'my_text_format',
+        ],
+        [
+          'label' => '16/07/2019',
+          'title' => 'Item 2',
+          'body' => 'Item 2 body',
+          'format' => 'my_text_format',
+        ],
+        [
+          'label' => '16/07/2019',
+          'title' => 'Item 3',
+          'body' => 'Item 3 body',
+          'format' => 'my_text_format',
+        ],
+      ],
+    ];
+
+    // Create node.
+    $node = Node::create($values);
+    $node->save();
+
+    // Verify the timeline is correctly rendered by ECL.
+    $display = EntityViewDisplay::collectRenderDisplay($node, 'default');
```

Comment:
Let's put a comment here explaining what we expect the timeline to be rendered like (given the formatter configuration).